### PR TITLE
FDA Drugs plugin

### PR DESCRIPTION
**Pull request by ctrl-schaff**
Plugin Reference: https://github.com/biothings/pending.api/issues/178

An example collection of drugs can be found here:
https://www.accessdata.fda.gov/scripts/cder/daf/index.cfm?event=overview.process&ApplNo=005010
The above link references a drug by the name of Demerol which the FDA has classified as `ApplNo: 005010`. The different types below are based off the various other properties the application had. Usually `Strength` and `Dosage Form/Route`, however each `ApplNo` will also have a corresponding `ProductNo`. For example the entry corresponding to `Applno: 005010` and `ProductNo: 002` corresponds to the drug application having `Strength: 50 MG/ML`.

Across the different tables provided by the FDA we've attempted to reconstruct the main table for each entry provided by the FDA. So for the following entry below:

```
"Drug Name" | "Active Ingredients" | "Strength" | "Dosage Form/Route" | "Marketing Status" | "TE Code" | "RLD" | "RS"
"DEMEROL" | "MEPERIDINE HYDROCHLORIDE" | "50MG/ML" | "INJECTABLE;INJECTION" | "Discontinued" | "None" | "No" | "No"
```

would correspond to the following document structure

```python
{
         "_id": [str],                                                
         "drug_name": [str],                                  
         "active_ingredients": [str],
         "strength": [str], 
         "dosage_form": [str],                                
         "marketing_status": [str],                           
         "therapeutic_equivalence": [str],                    
         "reference_listed_drug": [bool],                     
         "reference_standard": [bool]
}                                                                                  
```

### CODE REVIEW

File: plugins/fda_drugs/uploader.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import biothings.hub.dataload.uploader
+
+from biothings import config
+
+
+class FDA_DrugUploader(biothings.hub.dataload.uploader.BaseSourceUploader):
+    name = "fda_drugs"
+    __metadata__ = {
+        "src_meta": {
+            "url": "https://www.fda.gov/drugs/drug-approvals-and-databases/drugsfda-data-files",
```
Comment (newgene) [inline plugins/fda_drugs/uploader.py:21]: use this short url: "https://www.fda.gov/drugsatfda"

File: plugins/fda_drugs/uploader.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            "license_url": None,
```
Comment (newgene) [inline plugins/fda_drugs/uploader.py:22]: use this:
```
license": "CC0 1.0 Universal",
license_url": "https://creativecommons.org/publicdomain/zero/1.0/",
```

also add a `description` field (taken from their website):
```
description: "Drugs@FDA includes information about drugs, including biological products, approved for human use in the United States (see [FAQ](https://www.accessdata.fda.gov/scripts/cder/daf/index.cfm?event=faq.page)), but does not include information about FDA-approved products regulated by the Center for Biologics Evaluation and Research (for example, vaccines, allergenic products, blood and blood products, plasma derivatives, cellular and gene therapy products)."
```

File: plugins/fda_drugs/dumper.py [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+    NULL_MARKETING_STATUS,
+)
+
+
+logger = config.logger
+
+
+class FDA_DrugDumper(biothings.hub.dataload.dumper.LastModifiedHTTPDumper):
+    FDA_URL = "https://www.fda.gov"
+
```
Comment (everaldorodrigo) [inline plugins/fda_drugs/dumper.py:32]: 💅  could you remove this white space?

**Review by everaldorodrigo — APPROVED** [review]
@ctrl-schaff you could check the `Codacity` code analysis.

However, everything loos good to me.

File: plugins/fda_drugs/uploader.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+            <Strength>
+            <Dosage Form/Route>
+            <Marketing Status>
+            <TE Code>
+            <RLD>
+            <RS>
+        }
+        """
+        elasticsearch_mapping = {
+            "associatedWith": {
```
Comment (newgene) [inline plugins/fda_drugs/uploader.py:66]: why nested under `associatedWith`?

**Comment by newgene** [comment]
check above codacy issues too.

File: plugins/fda_drugs/uploader.py (hunk 4 above)
Comment (ctrl-schaff) [reply plugins/fda_drugs/uploader.py:66]: I think it was a holdover from another mapping I'd used. I'll remove it

**Comment by newgene** [comment]
@ctrl-schaff can you please also add some brief description to the comment of this PR, referencing the issue and probably posting a typical object example, describing the possible values for certain fields etc (or use a link as the reference), etc.

**Pull request merged**